### PR TITLE
multiple permit generation fix

### LINKED ISSUE #446: Fix multiple permit generation
Check this [issue](https://github.com/ubiquity/ubiquity-dollar/issues/643)

There are 2 permits generated for different networks:
1. https://github.com/ubiquity/ubiquity-dollar/issues/643#issuecomment-1569384273
2. https://github.com/ubiquity/ubiquity-dollar/issues/643#issuecomment-1607152588

What happened:
1. When the issue was closed the 1st time  the payout network was set to ethereum mainnet so the bot generated the [1st permit](https://github.com/ubiquity/ubiquity-dollar/issues/643#issuecomment-1569384273)
2. Then we updated the bot's config and switched to gnosis chain for payouts
3. When issue was closed the 2nd time the bot [couldn't find](https://github.com/ubiquity/ubiquibot/blob/development/src/handlers/payout/action.ts#L75) the permit URL for gnosis chain so generated a new permit

So on step 3 the bot should have been looking for permit for any chain, not the currently set in the config.

### PR DESCRIPTION
**Pull request by me505**
Resolves #446 

this issue is made to fix multiple permits being generated in a single issue

### CODE REVIEW

**Comment by me505** [comment]
@0xcodercrane can you review this

File: src/handlers/payout/action.ts [hunk 1]
```diff
@@ -74,9 +74,10 @@ export const handleIssueClosed = async () => {
   logger.info(`Posting a payout url to the issue, url: ${payoutUrl}`);
   const comment = `### [ **[ CLAIM ${priceInEth} ${tokenSymbol.toUpperCase()} ]** ](${payoutUrl})\n` + "```" + shortenRecipient + "```";
   const comments = await getAllIssueComments(issue.number);
-  const commentContents = comments.map((i) => i.body);
-  const exist = commentContents.find((content) => content.includes(comment));
-  if (exist) {
+  //const commentContents = comments.map((i) => i.body);
+  const permitComments = comments.filter((content) => content.body.includes("https://pay.ubq.fi?claim="));
+  const exist = permitComments.filter((content) => content.user.type == "Bot")
+  if (exist.length) {
```
Comment (0xcodercrane) [inline src/handlers/payout/action.ts:80]:
```suggestion
  const permitComments = comments.filter((content) => content.body.includes("https://pay.ubq.fi?claim=") && content.user.type == UserType.Bot);
    if (permitComments.length > 0) {
```

**Review by 0xcodercrane — CHANGES_REQUESTED** [review]

**Comment by 0xcodercrane** [comment]
@me505 you might need to import `UserType`

**Review by 0xcodercrane — APPROVED** [review]

File: src/handlers/payout/action.ts [hunk 2]
```diff
@@ -74,9 +74,8 @@ export const handleIssueClosed = async () => {
   logger.info(`Posting a payout url to the issue, url: ${payoutUrl}`);
   const comment = `### [ **[ CLAIM ${priceInEth} ${tokenSymbol.toUpperCase()} ]** ](${payoutUrl})\n` + "```" + shortenRecipient + "```";
   const comments = await getAllIssueComments(issue.number);
-  const commentContents = comments.map((i) => i.body);
-  const exist = commentContents.find((content) => content.includes(comment));
-  if (exist) {
+  const permitComments = comments.filter((content) => content.body.includes("https://pay.ubq.fi?claim=") && content.user.type == UserType.Bot);
```
Comment (pavlovcik) [inline src/handlers/payout/action.ts:77]: @rndquu this seems too brittle. 

We should actually parse the URL with the URL constructor, and then work backwards from there.

Comment (rndquu) [reply src/handlers/payout/action.ts:77]: Yes, parsing the URL with `URLSearchParams` is a better approach

This PR solves the original issue + we don't have plans of adding more query params besides `claim` and `network` so I don't think that refactoring to `URLSearchParams` is an urgent issue

> Besides, isn't it supposed to be `https://pay.ubq.fi/?claim=`

No, the generated URL is `https://pay.ubq.fi?claim=` so everything is fine

**Pull request merged**